### PR TITLE
[15.0][IMP] commission: Add new settlement period (All pending records) to settle in only one settlement

### PR DESCRIPTION
**Pull request by sergio-teruel**
cc @Tecnativa TT48574

ping @carlosdauden @victoralmau

### CODE REVIEW

File: commission/wizards/commission_make_settle.py [hunk 1]
```diff
@@ -97,7 +103,7 @@ def _prepare_settlement_line_vals(self, settlement, line):
             "settlement_id": settlement.id,
         }
 
-    def _get_agent_lines(self, date_to_agent):
+    def _get_agent_lines(self, agent, date_to_agent):
```
Comment (pedrobaeza) [inline commission/wizards/commission_make_settle.py:106]: Why do you need to add this argument? This is breaking possible extensions.

Comment (sergio-teruel) [reply commission/wizards/commission_make_settle.py:106]: I do not know... 
All modules that extends this method are changing the signature... So I change the signature in original method

Comment (pedrobaeza) [reply commission/wizards/commission_make_settle.py:106]: Ah, really? Then it can be considered a bug. Can you link one of those overrides?

Comment (sergio-teruel) [reply commission/wizards/commission_make_settle.py:106]: https://github.com/OCA/commission/blob/3a55b78b7f4773771577dc83d12d129a08147be1/account_commission/wizards/commission_make_settle.py#L17

**Review by carlosdauden — APPROVED** [review]

File: commission/wizards/commission_make_settle.py (hunk 1 above)
Comment (pedrobaeza) [reply commission/wizards/commission_make_settle.py:106]: Then it's definitively a bug here, so it's OK to change it indeed, thanks.

**Review by pedrobaeza — APPROVED** [review]
The only risk here is that you don't settle on time (accumulating several months), and you get everything in only one settlement.

**Review by victoralmau — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please fw-port it to 16 and 17

/ocabot merge minor

**Pull request merged**